blockchain/difficulty.go

```diff
@@ -218,7 +221,15 @@ func (b *BlockChain) findPrevTestNetDifficulty(startNode *blockNode) uint32 {
 // This function differs from the exported CalcNextRequiredDifficulty in that
 // the exported version uses the current best chain as the previous block node
 // while this function accepts any block node.
-func (b *BlockChain) calcNextRequiredDifficulty(lastNode *blockNode, newBlockTime time.Time) (uint32, error) {
+func (b *BlockChain) calcNextRequiredDifficulty(lastNode *blockNode,
+	newBlockTime time.Time) (uint32, error) {
+
+	// Emulate the same behavior as Bitcoin Core that for regtest there is
+	// no difficulty retargeting.
+	if b.chainParams.PoWNoRetargeting {
```

Comment:
Since this is only desired for Regtest (maybe simnet in btcd as well?), how about something like:

```
       if b.chainParams.Name == chaincfg.RegressionNetParams.Name {
```

instead of introducing a new field in `Params`?.

Reply:
IMO this makes it just way more explicit what's going on. See https://github.com/lightninglabs/neutrino/pull/256#discussion_r1162444890 :sweat_smile: